video/out/wayland_common.c

```diff
@@ -3229,20 +3229,21 @@ bool vo_wayland_valid_format(struct vo_wayland_state *wl, uint32_t drm_format, u
     // anyways, but might as well start from the most preferred tranche.
     struct vo_wayland_tranche *tranche;
     wl_list_for_each_reverse(tranche, &wl->tranche_list, link) {
-        bool supported_compositor_format = false;
         struct drm_format *formats = tranche->compositor_formats;
         for (int i = 0; i < tranche->num_compositor_formats; ++i) {
-            if (formats[i].format != drm_format)
+            if (drm_format != formats[i].format)
                 continue;
-            if (modifier == formats[i].modifier && modifier != DRM_FORMAT_MOD_INVALID)
-                return true;
-            supported_compositor_format = true;
-        }
-
-        if (supported_compositor_format && tranche->planar_formats) {
-            for (int i = 0; i < tranche->num_planar_formats; i++) {
-                if (drm_format == tranche->planar_formats[i])
+            if (modifier == formats[i].modifier) {
```

Comment:
Switching to early exit with `if (modifier != formats[i].modifier) continue;` would enable you to simplify further code. This also clarifies that the loop continuation is on purpose (same format may be in the tranche multiple times with different modifiers).